jsonrpc/debug.go

```diff
@@ -28,8 +38,44 @@ type StructLogs struct {
 	Storage map[string]string
 }
 
-func (d *Debug) TraceTransaction(hash ethgo.Hash) (*TransactionTrace, error) {
+func (d *Debug) TraceTransaction(hash ethgo.Hash, opts *TraceTransactionOptions) (*TransactionTrace, error) {
 	var res *TransactionTrace
-	err := d.c.Call("debug_traceTransaction", &res, hash)
+	err := d.c.Call("debug_traceTransaction", &res, hash, toTraceTransactionOpts(opts))
 	return res, err
 }
+
+func toTraceTransactionOpts(opts *TraceTransactionOptions) map[string]interface{} {
```

Comment:
Why does `json.Marshal` do not work here?

Reply:
It does. Fixed.